sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/info/GetSamplesInfo.groovy

```diff
@@ -42,10 +43,19 @@ class GetSamplesInfo implements GetSamplesInfoInput {
     try {
         def sampleCodes = samplesDataSource.fetchSampleCodesFor(projectCode, status)
         def sampleCodesToNames = infoDataSource.fetchSampleNamesFor(sampleCodes)
-        
-      output.samplesWithNames(projectCode, status, sampleCodesToNames)
+
+      List<Sample> samplesWithNames = buildSamples(sampleCodesToNames, status)
+      output.samplesWithNames(samplesWithNames)
     } catch (DataSourceException dataSourceException) {
       output.failedExecution(dataSourceException.getMessage())
     } 
   }
+
+  private static List<Sample> buildSamples(Map<String, String> codesToNames, Status status) {
+    List<Sample> samples = codesToNames.entrySet().stream()
+            .map({
+              return new Sample(it.key, it.value, status)
```

Comment:
```suggestion
            .map({ sampleCode, sampleName -> 
              return new Sample(sampleCode, sampleName, status)
```

Reply:
This will not work as we have one object here (it) being a Map.Entry